js/foam/dao/WebsocketDAO.js

```diff
@@ -0,0 +1,39 @@
+/* global CLASS */
+/* global JSONUtil */
+/* global io */
+CLASS({
+  package: 'foam.dao',
+  name: 'WebsocketDAO',
+  extendsModel: 'foam.dao.EasyClientDAO',
+  requires: [
+    'XHR',
+    'foam.core.dao.ClientDAO',
+  ],
+
+  methods: [
+    function forward(socket, messageName) {
+      var self = this;
+      socket.on(messageName, function(args) {
+        // TODO: a nicer way to do this?
+        var a = JSONUtil.parse(self.X, args[0]);
+        if (a.model_.name == self.model.name) {
```

Comment:
Using 'id' instead of 'name' will prevent false positives between Models with the same name in different packages.